pkg/server/auth.go

```diff
@@ -0,0 +1,82 @@
+package server
+
+import (
+	"context"
+	"net/url"
+	"sort"
+	"strings"
+
+	pkiutil "istio.io/istio/security/pkg/pki/util"
+)
+
+// authRequest will authenticate the request and authorize the CSR is valid for
+// the identity
+func (s *Server) authRequest(ctx context.Context, csrPEM []byte) (string, bool) {
+	caller, err := s.auther.Authenticate(ctx)
+	if err != nil {
+		// TODO: pass in logger with request context
+		s.log.Errorf("failed to authenticate request: %s", err)
+		return "", false
+	}
+
+	// request authentication has no identities, so error
+	if len(caller.Identities) == 0 {
+		s.log.Error("request sent with no identity")
+		return "", false
+	}
+
+	// return concatenated list of verified ids
+	identities := strings.Join(caller.Identities, ",")
+
+	csr, err := pkiutil.ParsePemEncodedCSR(csrPEM)
+	if err != nil {
+		s.log.Errorf("failed to decode CSR from %s: %s", identities, err)
+		return identities, false
+	}
+
+	// if the csr contains any other options set, error
+	if len(csr.DNSNames) > 0 || len(csr.IPAddresses) > 0 ||
```

Comment:
I retract my comment, it was caused by the lack of coffee

Reply:
:coffee: